moto/ec2/models.py

```diff
@@ -6675,7 +6675,8 @@ def delete_vpn_gateway(self, vpn_gateway_id):
     def detach_vpn_gateway(self, vpn_gateway_id, vpc_id):
         vpn_gateway = self.get_vpn_gateway(vpn_gateway_id)
         self.get_vpc(vpc_id)
-        detached = vpn_gateway.attachments.pop(vpc_id, None)
+        detached = vpn_gateway.attachments.get(vpc_id, None)
+        detached.state = "detached"
```

Comment:
The state should be assigned after the None-check (`if not detached`) - otherwise we'll get a uncaught exception

Reply:
Make sense